src/Configuration/src/KubernetesBase/KubernetesConfigMapProvider.cs

```diff
@@ -121,11 +134,52 @@ private void ProcessData(V1ConfigMap item)
             {
                 foreach (var data in item?.Data)
                 {
-                    configMapContents[NormalizeKey(data.Key)] = data.Value;
+                    // Should this check for application.{env.Environment}.json as well?
```

Comment:
Just to make sure I implement this in the expected manner, settings would be applied as followed:

Reply:
Maybe this should just look for starts with `appsettings` and ends with `json`? 

Let's look at the example in the linked issue:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: service-config
data:
  appsettings.json: |
    {
      "Test0": "Value0",
      "Test1": [
         {
            "Test2": "Value1"
         }
      ]
    }
```

This class is scoped to look at just one configmap, so is it more likely that multiple configmaps would be configured to mirror the multiple-file expectation we have with appsettings.json, or sould anybody want to combine those entries under a single configmap?

Reply:
Good point, I had overlooked the [KubernetesConfigurationBuilderExtensions.cs](https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Configuration/src/KubernetesBase/KubernetesConfigurationBuilderExtensions.cs#L45) already providing an environment specific configuration provider.
So to expand on the example from the issue to also include environment specific config settings as well, there would be 2 config maps:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: service-config
data:
  appsettings.json: |
    {
      "Test0": "Value0",
      "Test1": [
         {
            "Test2": "Value1"
         }
      ]
    }
```
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: service-config.demo
data:
  appsettings.demo.json: |
    {
      "Test1": [
         {
            "Test2": "Value2"
         }
      ]
    }
```
If the application is started with no EnvironmentName specified, the KubernetesConfigMapProvider bound to `service-config` will be the only provider to process data.
If the application is started with an EnvironmentName of `demo` specified, both KubernetesConfigMapProvider created will provide data, and the one bound to `service-config.demo` will override the former because it was added 2nd [here](https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Configuration/src/KubernetesBase/KubernetesConfigurationBuilderExtensions.cs#L45).

The expected values would be as follows then:
| EnvironmentName | Test0 value | Test1:0:Test2 value|
|----------------------|-------------|---------------------|
| Production             | Value 0       |  Value 1                 | 
| Demo                     | Value 0       |  Value 2                 |

Reply:
Yeah, that explanation matches what I would expect. What do you think of changing L138 to something along the lines of  `if the key ends with 'json' and value contains '{'` so that we aren't really caring what the key is so long as it's labeled like and looks to contain json? I don't have an overly strong opinion here, but it might be redundant to require the env name in both the configmap name and the key, especially if there's only one key